Libs/MRML/Core/vtkMRMLTableSQLiteStorageNode.cxx

```diff
@@ -0,0 +1,269 @@
+
+//----------------------------------------------------------------------------
+void vtkMRMLTableSQLiteStorageNode::PrintSelf(ostream& os, vtkIndent indent)
+{
+  vtkMRMLStorageNode::PrintSelf(os,indent);
+}
+
+//----------------------------------------------------------------------------
+bool vtkMRMLTableSQLiteStorageNode::CanReadInReferenceNode(vtkMRMLNode *refNode)
+{
+  return refNode->IsA("vtkMRMLTableNode");
+}
+
+//----------------------------------------------------------------------------
+int vtkMRMLTableSQLiteStorageNode::ReadDataInternal(vtkMRMLNode *refNode)
+{
+  std::string fullName = this->GetFullNameFromFileName();
+
+  if (fullName == std::string(""))
+    {
+    vtkErrorMacro("ReadData: File name not specified");
+    return 0;
+    }
+  vtkMRMLTableNode *tableNode = vtkMRMLTableNode::SafeDownCast(refNode);
+  if (tableNode == NULL)
+    {
+    vtkErrorMacro("ReadData: unable to cast input node " << refNode->GetID() << " to a table node");
+    return 0;
+    }
+
+  // Check that the file exists
+  if (vtksys::SystemTools::FileExists(fullName) == false)
+    {
+    vtkErrorMacro("ReadData: table file '" << fullName << "' not found.");
+    return 0;
+    }
+
+  std::string dbname = std::string("sqlite://") + fullName;
+  vtkSQLiteDatabase *database = vtkSQLiteDatabase::SafeDownCast( vtkSQLiteDatabase::CreateFromURL(dbname.c_str()));
+  if (!database->Open(this->GetPassword(), vtkSQLiteDatabase::USE_EXISTING))
+    {
+    vtkErrorMacro("ReadData: database file '" << fullName << "cannot be openned");
+    database->Delete();
+    return 0;
+    }
+
+  vtkSQLiteQuery *query = vtkSQLiteQuery::SafeDownCast( database->GetQueryInstance());
+  std::string queryString("select * from ");
+  queryString += std::string(this->TableName);
+  query->SetQuery(queryString.c_str());
+  query->Execute();
+
+  vtkSmartPointer<vtkRowQueryToTable> queryToTable = vtkSmartPointer<vtkRowQueryToTable>::New();
+  queryToTable->SetQuery(query);
+  queryToTable->Update();
+
+  vtkSmartPointer<vtkTable> table = queryToTable->GetOutput();
+
+  tableNode->SetAndObserveTable(table);
+
+  vtkDebugMacro("ReadData: successfully read table from file: " << fullName);
+
+  //database->Close();
+  query->Delete();
+  database->Delete();
+  return 1;
+}
+
+//----------------------------------------------------------------------------
+int vtkMRMLTableSQLiteStorageNode::WriteDataInternal(vtkMRMLNode *refNode)
+{
+  if (this->GetFileName() == NULL)
+    {
+    vtkErrorMacro("WriteData: file name is not set");
+    return 0;
+    }
+  std::string fullName = this->GetFullNameFromFileName();
+  if (fullName.empty())
+    {
+      vtkErrorMacro("WriteData: file name not specified");
+    return 0;
+    }
+
+  vtkMRMLTableNode *tableNode = vtkMRMLTableNode::SafeDownCast(refNode);
+  if (tableNode == NULL)
+    {
+    vtkErrorMacro("WriteData: unable to cast input node " << refNode->GetID() << " to a valid table node");
+    return 0;
+    }
+  if(this->TableName == "")
```

Comment:
This will probably not always work (even if this->TableName points to an empty string, its address will be probably different from the constant one).

TableName is const char, so first we need to check if it's NULL and if not then check if it's empty (by strcmp or by casting it to std::string and calling .empty).